hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMDirectoryCreateRequest.java

```diff
@@ -160,52 +178,86 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       } else if (omDirectoryResult == DIRECTORY_EXISTS_IN_GIVENPATH ||
           omDirectoryResult == NONE) {
         dirKeyInfo = createDirectoryKeyInfo(ozoneManager, omBucketInfo,
-            volumeName, bucketName, keyName, keyArgs, transactionLogIndex);
+            volumeName, bucketName, keyName, keyArgs, trxnLogIndex);
 
         omMetadataManager.getKeyTable().addCacheEntry(
             new CacheKey<>(omMetadataManager.getOzoneKey(volumeName, bucketName,
                 dirKeyInfo.getKeyName())),
-            new CacheValue<>(Optional.of(dirKeyInfo), transactionLogIndex));
+            new CacheValue<>(Optional.of(dirKeyInfo), trxnLogIndex));
+
+        omClientResponse = new OMDirectoryCreateResponse(omResponse.build(),
+            dirKeyInfo);
+      } else {
+        // omDirectoryResult == DIRECTORY_EXITS
+        // Check if this is a replay of ratis logs
+        String dirKey = omMetadataManager.getOzoneDirKey(volumeName,
```

Comment:
Instead of having DB read here, can we modify VerifyFilesInpath also have key info and check that here.
In this way, we don't do twice DB reads. 

```

else if (omMetadataManager.getKeyTable().isExist(dbDirKeyName)) {
        // Found a directory in the given path.
        // Check if this is actual directory or a directory in the given path
        if (dbDirKeyName.equals(dirNameFromDetails)) {
          return new Result[OMDirectoryResult.DIRECTORY_EXISTS, omMetadataManager.getKeyTable(dirNameFromDetails);
        }
```